Importing `AddressTrait` in the same file as `Module` macro breaks compilation

## Summary

```rust
error[E0252]: the name `AddressTrait` is defined multiple times
  --> module-system/module-implementations/sov-sequencer-registry/src/lib.rs:26:17
   |
11 | use sov_rollup_interface::AddressTrait;
   |     ---------------------------------- previous import of the trait `AddressTrait` here
...
26 | #[derive(Clone, ModuleInfo)]
   |                 ^^^^^^^^^^ `AddressTrait` reimported here
   ```

## Steps to reproduce

1. Define module with `Module` macro 
2. Define some function, which has `AddressTrait` generic.


## Expected Result

It compiles

## Actual Result

It does not compile


## Additional Information

--- a/full-node/db/sov-db/src/schema/tables.rs
+++ b/full-node/db/sov-db/src/schema/tables.rs
@@ -165,8 +165,8 @@ macro_rules! define_table_with_seek_key_codec {
 
         impl ::sov_schema_db::schema::KeyEncoder<$table_name> for $key {
             fn encode_key(&self) -> ::std::result::Result<::sov_rollup_interface::maybestd::vec::Vec<u8>, ::sov_schema_db::CodecError> {
-                use ::anyhow::Context;
-                use ::bincode::Options;
+                use ::anyhow::Context as _;
+                use ::bincode::Options as _;
 
                 let bincode_options = ::bincode::options()
                     .with_fixint_encoding()
@@ -178,8 +178,8 @@ macro_rules! define_table_with_seek_key_codec {
 
         impl ::sov_schema_db::schema::KeyDecoder<$table_name> for $key {
             fn decode_key(data: &[u8]) -> ::std::result::Result<Self, ::sov_schema_db::CodecError> {
-                use ::anyhow::Context;
-                use ::bincode::Options;
+                use ::anyhow::Context as _;
+                use ::bincode::Options as _;
 
                 let bincode_options = ::bincode::options()
                     .with_fixint_encoding()

--- a/module-system/sov-modules-macros/src/cli_parser.rs
+++ b/module-system/sov-modules-macros/src/cli_parser.rs
@@ -138,7 +138,7 @@ impl CliParserMacro {
             /// Borsh encode a transaction parsed from the CLI
             pub fn borsh_encode_cli_tx #impl_generics (cmd: CliTransactionParser #ty_generics) -> ::std::vec::Vec<u8>
             #where_clause {
-                use ::borsh::BorshSerialize;
+                use ::borsh::BorshSerialize as _;
                 match cmd {
                     #(#parse_match_arms)*
                     _ => panic!("unknown module name"),

--- a/module-system/sov-modules-macros/src/default_runtime.rs
+++ b/module-system/sov-modules-macros/src/default_runtime.rs
@@ -33,7 +33,7 @@ impl DefaultRuntimeMacro {
         Ok(quote::quote! {
         impl #impl_generics ::std::default::Default for #ident #type_generics #where_clause {
             fn default() -> Self {
-                use ::sov_modules_api::ModuleInfo;
+                use ::sov_modules_api::ModuleInfo as _;
 
                 Self {
                    #(#runtime_fn_body)*

--- a/module-system/sov-modules-macros/src/module_call_json_schema.rs
+++ b/module-system/sov-modules-macros/src/module_call_json_schema.rs
@@ -10,7 +10,7 @@ pub fn derive_module_call_json_schema(
     let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
 
     let tokens = quote::quote! {
-        use ::schemars::JsonSchema;
+        use ::schemars::JsonSchema as _;
 
         impl #impl_generics ::sov_modules_api::ModuleCallJsonSchema for #ident #type_generics #where_clause {
             fn json_schema() -> ::std::string::String {

--- a/module-system/sov-modules-macros/src/module_info.rs
+++ b/module-system/sov-modules-macros/src/module_info.rs
@@ -132,7 +132,6 @@ impl<'a> StructDef<'a> {
         let fn_dependencies = make_fn_dependencies(modules);
 
         Ok(quote::quote! {
-            use ::sov_modules_api::AddressTrait;
             impl #impl_generics ::std::default::Default for #ident #type_generics #where_clause{
 
                 fn default() -> Self {

--- a/module-system/sov-modules-macros/src/module_info.rs
+++ b/module-system/sov-modules-macros/src/module_info.rs
@@ -327,7 +327,7 @@ fn make_init_address(
             ),
         )),
         None => Ok(quote::quote! {
-            use ::sov_modules_api::digest::Digest;
+            use ::sov_modules_api::digest::Digest as _;
             let module_path = module_path!();
             let prefix = sov_modules_api::Prefix::new_module(module_path, stringify!(#struct_ident));
             let #field_ident : <#generic_param as sov_modules_api::Spec>::Address =

--- a/module-system/sov-modules-macros/tests/all_tests.rs
+++ b/module-system/sov-modules-macros/tests/all_tests.rs
@@ -3,6 +3,7 @@ fn module_info_tests() {
     let t = trybuild::TestCases::new();
     t.pass("tests/module_info/parse.rs");
     t.pass("tests/module_info/mod_and_state.rs");
+    t.pass("tests/module_info/use_address_trait.rs");
     t.compile_fail("tests/module_info/derive_on_enum_not_supported.rs");
     t.compile_fail("tests/module_info/field_missing_attribute.rs");
     t.compile_fail("tests/module_info/missing_address.rs");

--- a/module-system/sov-modules-macros/tests/module_info/use_address_trait.rs
+++ b/module-system/sov-modules-macros/tests/module_info/use_address_trait.rs
@@ -1,0 +1,11 @@
+//! Regression for <https://github.com/Sovereign-Labs/sovereign-sdk/issues/635>.
+
+use sov_modules_api::{AddressTrait, Context, ModuleInfo};
+
+#[derive(ModuleInfo)]
+struct TestModule<C: Context> {
+    #[address]
+    admin: C::Address,
+}
+
+fn main() {}

--- a/module-system/sov-modules-macros/tests/module_info/use_address_trait.rs
+++ b/module-system/sov-modules-macros/tests/module_info/use_address_trait.rs
@@ -1,5 +1,7 @@
 //! Regression for <https://github.com/Sovereign-Labs/sovereign-sdk/issues/635>.
 
+#![allow(unused_imports)]
+
 use sov_modules_api::{AddressTrait, Context, ModuleInfo};
 
 #[derive(ModuleInfo)]

